components/script/dom/webglshader.rs

```diff
@@ -114,15 +117,28 @@ impl WebGLShader {
                     // It could be interesting to retrieve the info log from the paint thread though
                     let msg = WebGLCommand::CompileShader(self.id, translated_source);
                     self.renderer.send(CanvasMsg::WebGL(msg)).unwrap();
-                    self.compilation_status.set(ShaderCompilationStatus::Succeeded);
+
+                    match self.parameter(constants::COMPILE_STATUS) {
```

Comment:
It's difficult for me to be confident about this after dealing in the past with some crappy android GPUs :)

Tests pass correctly without this. I haven't found an example that passes validation but fails to compile. I had an issue with this only when loading an OpenGL ES 3.0 context while doing some tests in osx, but the angle validator was set to 2.0

I don't want to prolong this PR, I restored original validation code. We can move this to another PR if needed.